src/AWSCredentials.jl

```diff
@@ -65,27 +78,25 @@ import Base: copy!
 Base.@deprecate copy!(dest::AWSCredentials, src::AWSCredentials) copyto!(dest, src)
 
 function AWSCredentials(;profile=nothing)
-
-    if haskey(ENV, "AWS_ACCESS_KEY_ID")
-
-        creds = env_instance_credentials()
-
-    elseif isfile(dot_aws_credentials_file()) || isfile(dot_aws_config_file())
-
-        creds = dot_aws_credentials(profile)
-
-    elseif haskey(ENV, "AWS_CONTAINER_CREDENTIALS_RELATIVE_URI")
-
-        creds = ecs_instance_credentials()
-
-    elseif localhost_maybe_ec2()
-
-        creds = ec2_instance_credentials()
-
-    else
-        error("Can't find AWS credentials!")
+    creds = nothing
+
+    # Define our search options
+    # NOTE: Ordering matters
+    functions = [
+        () -> env_instance_credentials(),
```

Comment:
This and the last one don't need to be wrapped in anonymous functions, they already have 0-arg methods.